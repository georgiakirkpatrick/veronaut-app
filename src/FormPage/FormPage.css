.empty {
    height: 0;
}

.FormPage {
    padding: 0 5vw 15vh;
    height: 90vh;
    overflow: scroll;
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    font: 16px/20px Poppins, sans-serif;
}

.FormPage button {
    font: 400 16px/22px Poppins, sans-serif;
}

.FormPage form {
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: flex-start;
}

.FormPage h1 {
    text-align: center;
    padding: 5vh 0;
    font: 300 30px Poppins, sans-serif;
}

.FormPage li {
    list-style: none;
}

.FormPage select {
    width: 100%;
}

.FormPage .NPFImages__preview-images {
    width: 100%;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: flex-start;
    margin: 0 auto 0;
}

.FormPage .NPFImages__preview-img { 
    width: 23%;
    margin: 5px 2% 0 0;
}

.FormPage .NPFImages__preview-img img { 
    width: 100%;
}

@media screen and (min-width: 768px) {
    .FormPage {
        padding: 0 7vw 15vh;
    }
}

@media screen and (min-width: 1024px) {
    .FormPage {
        padding: 0 7vw 15vh;
    }
}

@media screen and (min-width: 1024px) and (min-height: 1366px) {
    .FormPage {
        padding: 0 10vw 15vh;
        height: 93vh;
    }
}

@media screen and (min-width: 1440px) {
    .FormPage {
        padding: 5vh 15vw 10vh;
        height: 90vh;
    }
}

@media screen and (min-width: 2560px) {
    .FormPage {
        padding: 5vh 25vw 10vh;
    }
}