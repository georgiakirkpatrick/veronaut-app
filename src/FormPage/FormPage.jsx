import React from 'react'
import './FormPage.css'

const FormPage = ({
  title = '',
  children = '',
  nextError = ''
}) => {
  
  const formTitle = title ? <h1>{title}</h1> : <div className='empty' />

  return (
    <section className='FormPage'>
      {formTitle}
      
      <form id='form-page' autoComplete='off'>
        {children}
      </form>

      <div>
        {nextError}
      </div>
    </section>
  )
}

export default FormPage