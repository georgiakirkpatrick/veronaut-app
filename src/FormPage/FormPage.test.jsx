import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import FormPage from './FormPage'

describe('FormPage', () => {
    it ('renders without crashing', () => {
        render(<FormPage />)
        screen.debug()
    })

    it ('render the UI as expected', () => {
        const tree = render(<FormPage />)
        expect(tree).toMatchSnapshot()
    })
})