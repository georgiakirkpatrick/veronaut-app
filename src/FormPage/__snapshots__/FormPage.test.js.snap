// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPage render the UI as expected 1`] = `
<section
  className="FormPage"
>
  <div
    className="empty"
  />
  <form
    autoComplete="off"
    id="form-page"
  >
    
  </form>
  <div>
    
  </div>
</section>
`;
